bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/handler/ZonePlayerHandler.java

```diff
@@ -1468,6 +1477,45 @@ public boolean isOutputLevelFixed() {
         return stateMap.get("MusicSurroundLevel");
     }
 
+    public @Nullable String getCodec() {
+        String codec = stateMap.get("HTAudioIn");
+        switch (codec) {
```

Comment:
I think you should first check that the value is not null, just before the switch.

Reply:
Done.